spec.html

```diff
@@ -11785,8 +11785,10 @@ <h1>[[Set]] ( _P_, _V_, _Receiver_ )</h1>
           1. If Type(_P_) is String, then
             1. Let _numericIndex_ be ! CanonicalNumericIndexString(_P_).
             1. If _numericIndex_ is not *undefined*, then
-              1. Perform ? IntegerIndexedElementSet(_O_, _numericIndex_, _V_).
-              1. Return *true*.
+              1. If ! SameValue(_O_, _Receiver_) is *true*, then
+                1. Perform ? IntegerIndexedElementSet(_O_, _numericIndex_, _V_).
+                1. Return *true*.
+              1. If ! IsValidIntegerIndex(_O_, _numericIndex_) is *false*, return *true*.
```

Comment:
So if you have an integer index which is in bounds for O, but O and Receiver are different values, you'll now fall through to OrdinarySet. Is that intended?

Reply:
Honestly I don't remember what the behavior we got consensus for was, but this PR currently doesn't have interop wrt in-bounds/OOB accesses.

In-bounds falls through to OrdinarySet in SM and V8. OOB falls through to OrdinarySet in SM only. JSC currently always sets the value on the target instead of the receiver.

```javascript
// test-iieo-receiver-mismatch.js
let target = new Int8Array(1);
let receiver = [0];

// In-bounds
Reflect.set(target, 0, 1, receiver);
print('after in-bounds set:', target, receiver);

// OOB
Reflect.set(target, 2, 1, receiver);
print('after OOB set:', target, receiver);
```

```
$ eshost ./test-iieo-receiver-mismatch.js
#### ch
after in-bounds set: 0 1
after OOB set: 0 1,,1

#### jsc
after in-bounds set: 1 0
after OOB set: 1 0

#### sm
after in-bounds set: 0 1
after OOB set: 0 1,,1

#### v8
after in-bounds set: 0 1
after OOB set: 0 1

#### xs
after in-bounds set: 1 0
after OOB set: 1 0
```